pgjdbc/src/main/java/org/postgresql/core/v3/ConnectionFactoryImpl.java

```diff
@@ -116,21 +116,35 @@ public QueryExecutor openConnectionImpl(HostSpec[] hostSpecs, String user, Strin
     int connectTimeout = PGProperty.CONNECT_TIMEOUT.getInt(info) * 1000;
 
     // - the targetServerType setting
-    HostRequirement targetServerType;
+    HostRequirement requestedTargetServerType;
     try {
-      targetServerType =
+      requestedTargetServerType =
           HostRequirement.valueOf(info.getProperty("targetServerType", HostRequirement.any.name()));
```

Comment:
You should use targetServerTypeStr like the master branch.

Reply:
I had rebased this pacth and used targetServerTypeStr.